src/ring_theory/polynomial/hermite/gaussian.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2023 Luke Mantle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luke Mantle, Jake Levinson
+-/
+
+import ring_theory.polynomial.hermite.basic
+import analysis.special_functions.exp
+import analysis.special_functions.exp_deriv
+
+/-!
+# Hermite polynomials and Gaussians
+
+This file shows that the Hermite polynomial `hermite n` is (up to sign) the
+polynomial factor occurring in the `n`th derivative of a gaussian.
+
+## Results
+
+* `polynomial.deriv_gaussian_eq_hermite_mul_gaussian`:
+  The Hermite polynomial is (up to sign) the polynomial factor occurring in the
+  `n`th derivative of a gaussian.
+
+## References
+
+* [Hermite Polynomials](https://en.wikipedia.org/wiki/Hermite_polynomials)
+
+-/
+
+noncomputable theory
+open polynomial
+
+namespace polynomial
+
+/-- `hermite n` is (up to sign) the factor appearing in `deriv^[n]` of a gaussian -/
+lemma deriv_gaussian_eq_hermite_mul_gaussian (n : ℕ) (x : ℝ) :
+  deriv^[n] (λ y, real.exp (-(y^2 / 2))) x =
+  (-1 : ℝ)^n * aeval x (hermite n) * real.exp (-(x^2 / 2)) :=
+begin
+  induction n with n ih generalizing x,
+  { rw [function.iterate_zero_apply, pow_zero, one_mul, hermite_zero, C_1, map_one, one_mul] },
+  { replace ih : (deriv^[n] _) = _ := _root_.funext ih,
+    have deriv_gaussian : deriv (λ y, real.exp (-(y^2 / 2))) x = (-x) * real.exp (-(x^2 / 2)),
+    { simp [mul_comm, ← neg_mul] },
+    rw [function.iterate_succ_apply', ih, deriv_mul, deriv_const_mul, pow_succ (-1 : ℝ),
+        deriv_gaussian, hermite_succ, map_sub, map_mul, aeval_X, polynomial.deriv_aeval],
+    ring,
+    { simp_rw [aeval_def, eval₂_eq_eval_map],
+      apply polynomial.differentiable },
+    { simp_rw [aeval_def, eval₂_eq_eval_map],
+      apply differentiable.const_mul,
+      apply polynomial.differentiable },
+    { simp } }
```

Comment:
These side goals go away if you work with `has_deriv_at` instead of `deriv`:
```lean
  induction n with n ih generalizing x,
  { rw [function.iterate_zero_apply, pow_zero, one_mul, hermite_zero, C_1, map_one, one_mul] },
  { replace ih : (deriv^[n] _) = _ := _root_.funext ih,
    have has_deriv_at_gaussian : ∀ x,
      has_deriv_at (λ y, real.exp (-(y^2 / 2))) ((-x) * real.exp (-(x^2 / 2))) x,
    { intro x,
      simpa [mul_div_cancel_left x (two_ne_zero), neg_mul, mul_comm]
        using (((has_deriv_at_id' x).pow 2).div_const 2).neg.exp },
    rw [function.iterate_succ_apply', ih],
    simp_rw [aeval_def, eval₂_eq_eval_map],
    have := ((polynomial.has_deriv_at _ _).const_mul ((-1 : ℝ) ^ n)).mul (has_deriv_at_gaussian x),
    refine this.deriv.trans _,
    sorry },
```

Reply:
Actually also `deriv_const_mul_field` doesn't generate a side goal, I didn't notice it before (versus `deriv_const_mul`).

Reply:
The new version is pretty close to this, let's just go with it.